### ISSUE TITLE
Server-side search

### ISSUE THREAD

**Issue by phil-s** [alphapapa/ement.el#286]
It would be nice to have support for [server-side search][1] in Ement.el.

@alphapapa​ commented in chat:

> What's needed is support for the events API so that when we show a message from a search result we can show the surrounding context.

(The [Event Context][2] provided with each result (if requested) contains those surrounding `events_before` and `events_after`.)

[1]: https://spec.matrix.org/latest/client-server-api/#server-side-search
[2]: https://spec.matrix.org/latest/client-server-api/#post_matrixclientv3search_response-200_event-context

**Comment by alphapapa** [comment]
Yeah, that much could be done quickly.  What I meant to say is that we need to be able to jump to an arbitrary event in a room and then go forward/back in the timeline, loading other events as needed, and possibly integrating them with the timeline we already have for the room.  If the search feature were limited to just seeing the matching event and a few on either side of it, it would feel very limiting.  So we need more comprehensive support for the events API, integrating it with existing functionality (that would also facilitate bookmarking events in rooms and then jumping back to them).
